scripts/use-relative-md-links.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * A Markdownlint Plugin that checks whether the relative link urls are file-relative instead of
+ * directory-relative to ensure that mkdocs validates the links.
+ */
+
+// All external urls should have a scheme, i.e. http://, https:// or mailto:
+const externalUrl = /^[a-z0-9+.-]+:/;
+// linking within the same file is just (#<anchor-name>)
+const anchor = /^#/;
+const markdownFile = /.md/;
+const svg = /.svg/;
+
+module.exports = {
+	names: ['NAUTOBOTMD001', 'nautobotmd.filerelativelinks'],
+	description: 'All relative links should be file-relative',
+	tags: ['links'],
+	function: function rule(params, onError) {
+		params.tokens
+			.filter(function filterToken(token) {
+				return token.type === 'inline';
+			})
+			.forEach(function forToken(inline) {
+				inline.children
+					.filter(function filterChild(child) {
+						return child.type === 'link_open';
+					})
+					.forEach(function forToken(link) {
+						if (link.attrs[0] && link.attrs[0][0] === 'href') {
+							const href = link.attrs[0][1];
+
+							let isMarkdownFile = markdownFile.test(href.toLowerCase());
+							let isExternalUrl = externalUrl.test(href.toLowerCase());
+							// There is one relative link in the docs that points to an SVG file.
```

Comment:
We have relative references to other file types too, such as the various jpg (ew) and png images in `docs/media/`. I'm guessing those aren't checked by this rule since they're generally embedded media rather than hyperlinks, but I wonder if it would be cleaner or more compatible in general if instead of checking specifically for `.md` and `.svg` if we just checked generally for a file extension, something like `/\.[a-z]+$/`?

Reply:
Correct, it only looks for markdown links, it does not check images. The only reason that the .svg was flagged is that it is a link to an image file.

Reply:
I think that may not catch things we believe it should then.  I think a better option is to omit checking this rule on the svg line instead. I will see if this is possible.